hls4ml/templates/vivado/build_prj.tcl

```diff
@@ -162,9 +162,11 @@ if {$opt(reset)} {
     open_solution "solution1"
 }
 catch {config_array_partition -maximum_size 4096}
-config_compile -name_max_length 60
+config_compile -name_max_length 80
 set_part $part
+config_schedule -enable_dsp_full_reg=false
 create_clock -period $clock_period -name default
+set_clock_uncertainty 12.5% default
```

Comment:
I would recommend making the clock uncertainty configurable (and maybe sticking with the Vitis_HLS default if not set, though one could argue otherwise if we generally meet stricter timing).

Reply:
Added the ability to tweak the clock uncertainty in the latest commit. Use with `yaml_config['ClockUncertainty'] = '20%'`. Note that specifying a value without `%` (and quotes) will be interpreted as an absolute value in ns to use as uncertainty (as per UG902), so to specify a percentage, you have to put it in quotes to make it a valid python code.